src/components/ChannelPreview/hooks/useChannelPreviewInfo.ts

```diff
@@ -21,28 +21,29 @@ export const useChannelPreviewInfo = <
 ) => {
   const { channel, overrideImage, overrideTitle } = props;
 
-  const { client } = useChatContext<StreamChatGenerics>('ChannelPreview');
-  const [displayTitle, setDisplayTitle] = useState(getDisplayTitle(channel, client.user));
-  const [displayImage, setDisplayImage] = useState(getDisplayImage(channel, client.user));
+  const { client } = useChatContext<StreamChatGenerics>('useChannelPreviewInfo');
+  const [displayTitle, setDisplayTitle] = useState(
+    overrideTitle || getDisplayTitle(channel, client.user),
+  );
+  const [displayImage, setDisplayImage] = useState(
+    overrideImage || getDisplayImage(channel, client.user),
+  );
```

Comment:
```suggestion
  const [displayTitle, setDisplayTitle] = useState(
    () => overrideTitle || getDisplayTitle(channel, client.user),
  );
  const [displayImage, setDisplayImage] = useState(
    () => overrideImage || getDisplayImage(channel, client.user),
  );
```

I would use function useState initializer here to prevent running and immediately discarding the result of `getDisplayImage()`

Reply:
True

Reply:
Done